compact_test.go

```diff
@@ -312,44 +300,69 @@ func TestLeveledCompactor_plan(t *testing.T) {
 			},
 			expected: []string{"7", "8"},
 		},
-		// For overlapping blocks.
-		{
+		//|--------------|
+		//              |----------------|
+		//			         |--------------|
+		"Overlapping blocks 1": {
 			metas: []dirMeta{
 				metaRange("1", 0, 20, nil),
 				metaRange("2", 19, 40, nil),
 				metaRange("3", 40, 60, nil),
 			},
 			expected: []string{"1", "2"},
 		},
-		{
+		//|--------------|
+		//               |--------------|
+		//			 |--------------|
+		"Overlapping blocks 2": {
 			metas: []dirMeta{
 				metaRange("1", 0, 20, nil),
 				metaRange("2", 20, 40, nil),
 				metaRange("3", 30, 50, nil),
 			},
 			expected: []string{"2", "3"},
 		},
-		{
+		//|--------------|
+		//        |--------------|
+		//		 |--------------|
+		"Overlapping blocks 3": {
 			metas: []dirMeta{
 				metaRange("1", 0, 20, nil),
-				metaRange("2", 10, 40, nil),
-				metaRange("3", 30, 50, nil),
+				metaRange("2", 10, 30, nil),
+				metaRange("3", 20, 40, nil),
 			},
 			expected: []string{"1", "2", "3"},
 		},
-		{
+		//|--------------|
+		//              |-------------------------------|
+		//		 |--------------|
+		//			        |--------------|
+		"Overlapping blocks 4": {
 			metas: []dirMeta{
-				metaRange("5", 0, 360, nil),
-				metaRange("6", 340, 560, nil),
-				metaRange("7", 360, 420, nil),
-				metaRange("8", 420, 540, nil),
+				metaRange("5", 0, 20, nil),
+				metaRange("6", 19, 61, nil),
+				metaRange("7", 20, 40, nil),
+				metaRange("8", 40, 60, nil),
 			},
 			expected: []string{"5", "6", "7", "8"},
 		},
+		//|--------------|
+		//        |--------------|
+		// 				  |--------------|
+		//       				  |--------------|
+		"Overlapping blocks 5": {
+			metas: []dirMeta{
```

Comment:
Is there any use case as this https://github.com/prometheus/tsdb/issues/554 actually?

Reply:
i don't see why not. the added blocks can have any ranges